### DISCUSSION TITLE
run-workflow-step-or-job-based-on-file-changes-github-actions

### DISCUSSION THREAD

**Discussion by giscus** [thomasnotfound/how.wtf#231] [Blog Posts]
# run-workflow-step-or-job-based-on-file-changes-github-actions

How to catch errors in Boto3 with Python

https://how.wtf/run-workflow-step-or-job-based-on-file-changes-github-actions.html

**Comment by JoshMcCullough** [comment]
Thank you! Note that the "Run job ..." and "Run step ..." sections are backwards ("run job" shows an example for running steps and "run step" shows an example for running jobs).

**Reply by thomasgtaylor** [reply]
Thank you for reading! Also, great catch – I updated the article.

**Comment by jvenus99** [comment]
I had that warning when i add the if condition(if: steps.changes.outputs.infra == 'true'): Context access might be invalid

**Reply by ghost** [reply]
Thanks for reporting this. Did the flow continue to work regardless of the warning?

**Comment by trusktr** [comment]
GitHub Actions already has [paths filters](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore).

F.e.:

```yml
on:
  push:
    paths:
      - '**.js'
```

Why is [`dorny/paths-filter`](https://github.com/dorny/paths-filter) better? Seems like the reasoning in `dorny/paths-filter` is that the config can be in a single file. With GitHub Actions, the same can be done by making multiple files, right? Maybe that's even preferable.

**Reply by trusktr** [reply]
Ah, looks like GitHub Actions mentions this about their path filters:

> [!Note]
> You should not use path filtering to skip workflow runs if the workflow has been configured to be required. For more information, see "[Required workflows](https://docs.github.com/en/actions/using-workflows/required-workflows)."
> 
> If a workflow is skipped due to path filtering, [branch filtering](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore), or a [commit message](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs), then checks associated with that workflow will remain in a "Pending" state. A pull request that requires those checks to be successful will be blocked from merging.

So, using `dorny/paths-filter` should allow the action to always run, but return early if files don't match.
